test/conformance/runtime/util.go

```diff
@@ -89,16 +93,19 @@ func runtimeInfo(
 	return names, &ri, err
 }
 
-func splitOpts(opts ...interface{}) ([]ServiceOption, []pkgTest.RequestOption) {
+func splitOpts(opts ...interface{}) ([]ServiceOption, []pkgTest.RequestOption, error) {
 	serviceOpts := []ServiceOption{}
 	reqOpts := []pkgTest.RequestOption{}
 	for _, opt := range opts {
-		switch opt.(type) {
+		switch t := opt.(type) {
 		case ServiceOption:
 			serviceOpts = append(serviceOpts, opt.(ServiceOption))
 		case pkgTest.RequestOption:
 			reqOpts = append(reqOpts, opt.(pkgTest.RequestOption))
+		default:
+			return nil, nil, fmt.Errorf("invalid option type: %v", t)
```

Comment:
I'm not sure this will do what you want, I think you probably want `... %T", opt)`

Reply:
Yeah that does look better... fixed.

Example of bad type being passed:
    header_test.go:113: Error fetching runtime info: invalid option type: http.Header